great_expectations/expectations/metrics/util.py

```diff
@@ -608,26 +615,44 @@ def column_reflection_fallback(
             # noinspection PyUnresolvedReferences
             if dialect.name.lower() == GXSqlDialect.REDSHIFT:
                 # Redshift needs temp tables to be declared as text
-                query: Select = (
+                query = (
                     sa.select([sa.text("*")]).select_from(sa.text(selectable)).limit(1)
                 )
             else:
-                query: Select = (
-                    sa.select([sa.text("*")]).select_from(selectable).limit(1)
-                )
+                query = sa.select([sa.text("*")]).select_from(selectable).limit(1)
         result_object = sqlalchemy_engine.execute(query)
         # noinspection PyProtectedMember
         col_names: List[str] = result_object._metadata.keys
         col_info_dict_list = [{"name": col_name} for col_name in col_names]
     return col_info_dict_list
 
 
+@overload
 def get_dbms_compatible_column_names(
-    column_names: Union[List[str], str],
-    batch_columns_list: List[Union[str, quoted_name]],
```

Comment:
I'm pretty sure this was correct. There may be some incorrectly annotated variables elsewhere. @alexsherstinsky would know more.

Reply:
@Kilo59 Yes, definitely need `quoted_name` (it is an `SQLAlchemy` object).  Actually, it would be best to leave the `get_dbms_compatible_column_names()` as it was -- all of the argument types were very meticulously defined and tested.  Thanks. /cc @NathanFarmer

Reply:
The issue was the annotation was not valid. But @Nathan provided the import path for the actual object so I will update and add it.

Reply:
Updated with a valid annotation for `sqlalchemy.sql.quoted_name`